org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/format/DOMAttributeFormatter.java

```diff
@@ -77,6 +78,18 @@ public void formatAttribute(DOMAttr attr, int prevOffset, boolean singleAttribut
 				removeLeftSpaces(delimiterOffset, attrValueStart, edits);
 			}
 		}
+
+		// replace quotation marks according to settings
+		String originalValue = attr.getOriginalValue();
+		if (getEnforceQuoteStyle() == EnforceQuoteStyle.preferred
+				&& originalValue != null
+				&& originalValue.charAt(0) != getQuotationAsChar()) {
+			// replace current quote with preferred quote
+			formatterDocument.replaceQuoteWithPreferred(attr.getNodeAttrValue().getStart(),

```

Comment:
It seems that you assume that there are quote but what about invalid attributs

name=123"
name="123

Please write test with those usecases.